packages/wallet/src/redux/protocols/reducer-helpers.ts

```diff
@@ -88,6 +89,45 @@ export function sendOpponentConcluded(sharedData: SharedData): SharedData {
   return newSharedData;
 }
 
+export function sendCommitments(
+  sharedData: SharedData,
+  processId: string,
+  channelId: string,
+  protocolLocator?: string,
+): SharedData {
+  const channel = getExistingChannel(sharedData, channelId);
+  const { participants, ourIndex } = channel;
+  const messageRelay = sendCommitmentsReceived(
+    nextParticipant(participants, ourIndex),
+    processId,
+    channel.commitments,
+    protocolLocator,
+  );
+  return queueMessage(sharedData, messageRelay);
+}
+
+export function checkCommitments(
+  sharedData: SharedData,
+  turnNum: number,
+  commitments: Commitments,
+): SharedData {
+  // We don't bother checking "stale" commitments -- those whose turnNum does not
+  // exceed the current turnNum.
+
+  commitments
+    .filter(signedCommitment => signedCommitment.commitment.turnNum > turnNum)
+    .map(signedCommitment => {
+      const result = checkAndStore(sharedData, signedCommitment);
+      if (result.isSuccess) {
+        sharedData = result.store;
+      } else {
+        throw new Error('Unable to validate commitment');
```

Comment:
Should we rewrite this to return `{ isSuccess: false, reason: `Unable to validate commitment: ${result.reason}` }` instead?

Reply:
This was a function you wrote ;)
Are you suggesting the return type be `{sharedData:SharedData, isSuccess:boolean, reason?:string}`?

Reply:
Maybe? I'm not really up to speed on our approach to handling errors.

Reply:
Unfortunately I don't think we have a unified approach at the moment 😢